llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -56,24 +67,23 @@ void llvm::getSYCLDeviceRequirements(
         std::vector<uint32_t>(Values.begin(), Values.end());
   }
 
-  std::optional<llvm::SmallVector<int64_t, 3>> ReqdWorkGroupSize;
+  std::optional<llvm::SmallVector<size_t, 3>> ReqdWorkGroupSize;
```

Comment:
I would preserve `uint64_t` here. The tricky thing here is that we are trying to match `size_t` selected by our SYCL compiler for user's application. Strictly speaking, it may not be the same `size_t` which was selected by a compiler used to build the toolchain itself.

Having fixed-width type here at least makes it explicit and obvious about which size is supported by the toolchain and makes it independent of how toolchain is built.